pandas/core/series.py

```diff
@@ -4564,6 +4565,96 @@ def replace(
             method=method,
         )
 
+    @Substitution(
+        klass="Series",
+        type_sub="",
+        max_cols_sub="",
+        examples_sub=(
+            """
+            >>> int_values = [1, 2, 3, 4, 5]
+            >>> text_values = ['alpha', 'beta', 'gamma', 'delta', 'epsilon']
+            >>> s = pd.Series(text_values, index=int_values)
+            >>> s.info()
+            <class 'pandas.core.series.Series'>
+            Int64Index: 5 entries, 1 to 5
+            Series name: None
+            Non-Null Count  Dtype
+            --------------  -----
+            5 non-null      object
+            dtypes: object(1)
+            memory usage: 80.0+ bytes
+
+            Prints a summary excluding information about its values:
+
+            >>> s.info(verbose=False)
+            <class 'pandas.core.series.Series'>
+            Int64Index: 5 entries, 1 to 5
+            dtypes: object(1)
+            memory usage: 80.0+ bytes
+
+            Pipe output of Series.info to buffer instead of sys.stdout, get
+            buffer content and writes to a text file:
+
+            >>> import io
+            >>> buffer = io.StringIO()
+            >>> s.info(buf=buffer)
+            >>> s = buffer.getvalue()
+            >>> with open("df_info.txt", "w",
+            ...           encoding="utf-8") as f:  # doctest: +SKIP
+            ...     f.write(s)
+            260
+
+            The `memory_usage` parameter allows deep introspection mode, specially
+            useful for big Series and fine-tune memory optimization:
+
+            >>> random_strings_array = np.random.choice(['a', 'b', 'c'], 10 ** 6)
+            >>> s = pd.Series(np.random.choice(['a', 'b', 'c'], 10 ** 6))
+            >>> s.info()
+            <class 'pandas.core.series.Series'>
+            RangeIndex: 1000000 entries, 0 to 999999
+            Series name: None
+            Non-Null Count    Dtype
+            --------------    -----
+            1000000 non-null  object
+            dtypes: object(1)
+            memory usage: 7.6+ MB
+
+            >>> s.info(memory_usage='deep')
+            <class 'pandas.core.series.Series'>
+            RangeIndex: 1000000 entries, 0 to 999999
+            Series name: None
+            Non-Null Count    Dtype
+            --------------    -----
+            1000000 non-null  object
+            dtypes: object(1)
+            memory usage: 55.3 MB"""
+        ),
+        see_also_sub=(
+            """
+            Series.describe: Generate descriptive statistics of Series.
+            Series.memory_usage: Memory usage of Series."""
+        ),
+    )
+    @doc(SeriesInfo.to_buffer)
```

Comment:
SeriesInfo.to_buffer doesn't have a docstring. so this doesn't render.

```
>>> help(pd.Series.info)
Help on function info in module pandas.core.series:

info(self, verbose: Union[bool, NoneType] = None, buf: Union[IO[str], NoneType] = None, max_cols: Union[int, NoneType] = N
one, memory_usage: Union[bool, str, NoneType] = None, null_counts: Union[bool, NoneType] = None) -> None

>>>
```

and wouldn't have memory_usage, max_cols, and null_counts parameters anyway?

(as an aside there appears to be a few issues with DataFrame.info docstring on master, such as alignment of console output and rogue data parameter. Not sure if always like this or from recent refactors, so if you get time, it would be great if can you check that out)

Reply:
Regarding ``null_counts`` - does it mean that we do not need series info without non-null counts?

Reply:
> (as an aside there appears to be a few issues with DataFrame.info docstring on master, such as alignment of console output and rogue data parameter. Not sure if always like this or from recent refactors, so if you get time, it would be great if can you check that out)

I noticed not only here, but in couple of other places, that indentation gets bad, when using this kind of construct:
```
        %(max_cols_sub)s
```
I never touched the docstring, so probably @MarcoGorelli can comment on the rendering issue.

Reply:
Please note that I just added dedent in some parameters docs, which make ``info`` docstrings render better, without extra indentation.